### PR TITLE
UX improvements for the previewer app

### PR DESCRIPTION
**Pull request by hbons**
for #55.

### CODE REVIEW

File: tools/FigmaSharpApp/Views/DocumentWindowController.cs [hunk 1]
```diff
@@ -60,12 +54,24 @@ public DocumentWindowController (IntPtr handle) : base (handle)
...
+
+			CGRect frame = Window.Frame;
+			ShouldCascadeWindows = true;
+
+			frame.Width  = NSScreen.MainScreen.Frame.Width  * 0.8f;
+			frame.Height = NSScreen.MainScreen.Frame.Height * 0.8f;
+
+			Window.SetFrame(frame, display: true);
+
+			if (NSApplication.SharedApplication.Windows.Length == 3) // Magic number :(
```
Comment (netonjm) [inline tools/FigmaSharpApp/Views/DocumentWindowController.cs]: what's the magic on this?

File: tools/FigmaSharpApp/Views/DocumentWindowController.cs [hunk 2]
```diff
@@ -89,24 +95,49 @@ public void EnableButtons (bool enable)
...
 
-			menu.AddItem ("1", "FigmaSharp.Cocoa 0.0.1", DateTime.Now);
-			menu.AddItem ("2", "FigmaSharp.Cocoa 0.0.2", DateTime.Now);
-			menu.AddItem ("3", "FigmaSharp.Cocoa 0.0.3", DateTime.Now);
-			menu.AddItem ("4", DateTime.Now);
-			menu.AddItem ("5", DateTime.Now.AddDays (-7));
-			menu.AddItem ("6", DateTime.Now.AddDays (-14));
 
-			menu.UseAsVersionsMenu ();
+			new Thread(() => {
```
Comment (netonjm) [inline tools/FigmaSharpApp/Views/DocumentWindowController.cs]: Better use: Task.Run (() =>

File: tools/FigmaSharpApp/Views/DocumentViewController.cs [hunk 3]
```diff
@@ -52,27 +55,51 @@ public partial class DocumentViewController : NSViewController
...
-			Load (version_id: versionId, page_id: pageId, startPage:0);
+			ToggleSpinnerState(toggle_on: true);
+			Load(version_id: versionId, page_id: pageId, startPage:0);
 		}
 
 		public void OnInitialize ()
 		{
+			var windowDelegate = new WindowDelegate();
+
+			windowDelegate.WindowFocused += delegate {
```
Comment (netonjm) [inline tools/FigmaSharpApp/Views/DocumentViewController.cs]: did you tried using the NSNotificationCenter?

File: tools/FigmaSharpApp/Views/DocumentViewController.cs [hunk 4]
```diff
@@ -133,9 +158,50 @@ void Load (string version_id = "", string page_id = "", int startPage = 0)
 					////NOTE: some toolkits requires set the real size of the content of the scrollview before position layers
 					scrollview.AdjustToContent ();
 
+
+					NSScrollView scrollView = (scrollview.NativeObject as NSScrollView);
+					NSView windowView = windowController.Window.ContentView;
```
Comment (netonjm) [inline tools/FigmaSharpApp/Views/DocumentViewController.cs]: window could be null here

File: tools/FigmaSharpApp/Views/DocumentViewController.cs [hunk 5]
```diff
@@ -133,9 +158,50 @@ void Load (string version_id = "", string page_id = "", int startPage = 0)
...
+					NSView documentView = (scrollView.DocumentView as NSView);
+
+
+					// Center the document
+					var posX = 0.5 * (windowView.Frame.Width  - documentView.Frame.Width);
+					var posY = 0.5 * (windowView.Frame.Height - documentView.Frame.Height);
+
+					NSView wrapper = new NSView(new CGRect(0, 0, windowView.Frame.Width, windowView.Frame.Height));
```
Comment (netonjm) [inline tools/FigmaSharpApp/Views/DocumentViewController.cs]: NSView wrapper = new NSView(windowView.Bounds);

File: tools/FigmaSharpApp/Views/OpenLocationViewController.cs [hunk 6]
```diff
@@ -66,6 +66,12 @@ public override void ViewDidLoad ()
 			OpenButton.Enabled = CheckFormIsFilled ();
 		}
 
+		public override void ViewDidAppear()
+		{
+			base.ViewDidAppear();
+			OpenLocationWindow.Window.Center();
```
Comment (netonjm) [inline tools/FigmaSharpApp/Views/OpenLocationViewController.cs]: check window null here

File: tools/FigmaSharpApp/Views/DocumentWindowController.cs (hunk 1 above)
Comment (hbons) [reply tools/FigmaSharpApp/Views/DocumentWindowController.cs]: I want to center the first window that appears. For some reason, when only one window is open, this returns 3…

**Pull request merged**